lazy_dataset/core.py

```diff
@@ -1841,79 +1833,161 @@ def __iter__(self):
                 yield example
 
 
+class Bucket:
+    def __init__(self, init_example, batch_size):
+        """
+        Base Bucket performing standard batching.
+        You can inherit this class to define sophisticated buckets.
+
+        Args:
+            init_example: first example in the batch
+            batch_size: number of examples in a batch
+        """
+        self.data = [init_example]
+        self.batch_size = batch_size
+
+    def is_completed(self):
+        return len(self.data) >= self.batch_size
+
+    def assess(self, example):
+        return not self.is_completed()
```

Comment:
Can you raise an `NotImplementError`? Forgetting to overwrite this function will otherwise result in an fallback of dynamic batching to batching.